ext/DifferentiationInterfaceReverseDiffExt.jl

```diff
@@ -34,6 +30,15 @@ function DI.value_and_pullback!(
     return y, dx
 end
 
+function DI.value_and_pullback!(
+    _dx, backend::AutoReverseDiff, f, x::X, dy::Y, extras::Nothing=nothing
+) where {X<:Number,Y}
+    x_array = [x]
+    dx_array = [zero(x)]
+    y, dx_array = DI.value_and_pullback!(dx_array, backend, f ∘ only, x_array, dy, extras)
```

Comment:
Why do we call `only` on the output of `f`?

Reply:
no we call `only` on the vector input, to turn it into a scalar that `f` can eat

Reply:
you misread the composition syntax